packages/angular/ssr/src/routes/ng-routes.ts

```diff
@@ -272,26 +374,30 @@ export async function getRoutesFromAngularRouterConfig(
  *  - https://github.com/angular/angular/blob/6882cc7d9eed26d3caeedca027452367ba25f2b9/packages/platform-server/src/http.ts#L44
  * @param manifest - An optional `AngularAppManifest` that contains the application's routing and configuration details.
  * If not provided, the default manifest is retrieved using `getAngularAppManifest()`.
- *
+ * @param invokeGetPrerenderPaths - A boolean flag indicating whether to invoke `getPrerenderPaths` for parameterized SSG routes
+ * to handle prerendering paths. Defaults to `false`.
  * @returns A promise that resolves to a populated `RouteTree` containing all extracted routes from the Angular application.
  */
 export async function extractRoutesAndCreateRouteTree(
   url: URL,
   manifest: AngularAppManifest = getAngularAppManifest(),
+  invokeGetPrerenderPaths = false,
```

Comment:
Question: When would we choose to set `invokeGetPrerenderPaths = false`? Is that for the devserver use case which uses SSR for SSG routes? If so, I'm wondering if this should be a slightly more generic `allowPrerendering` which controls both:
1. Whether an SSG route is rendered as SSG or SSR.
2. Whether `getPrerenderPaths` is invoked.

Is that a better abstraction here? Is 1. even an issue in this context?

Reply:
`invokeGetPrerenderPaths` should be `false` in almost all the case, the only time this needs to be `true` is when the this method is being invoked to generate the manifest. 

I prefer the parameter name to be called `invokeGetPrerenderPaths` as it's clear and `allowPrerendering` can be misleading.